### PR TITLE
[Optimization] Remove case-of-case

### PR DESCRIPTION
**Pull request by effectfully**
This removes the case-of-case optimization entirely due to it having multiple sources of exponentiality OOming users' machines. See #6183

### CODE REVIEW

File: plutus-core/plutus-ir/src/PlutusIR/Compiler.hs [hunk 1]
```diff
@@ -136,14 +135,12 @@ simplifierIteration suffix = do
...
   hints <- view (ccOpts . coInlineHints)
   preserveLogging <- view (ccOpts . coPreserveLogging)
-  cocConservative <- view (ccOpts . coCaseOfCaseConservative)
   rules <- view ccRewriteRules
   ic <- view (ccOpts . coInlineConstants)
 
   pure $ P.NamedPass ("simplifier" ++ suffix) $ fold
       [ mwhen (opts ^. coDoSimplifierUnwrapCancel) $ Unwrap.unwrapCancelPass tcconfig
       , mwhen (opts ^. coDoSimplifierCaseReduce) $ CaseReduce.caseReducePass tcconfig
-      , mwhen (opts ^. coDoSimplifierCaseReduce) $ CaseOfCase.caseOfCasePassSC tcconfig binfo cocConservative noProvenance
```
Comment (effectfully) [inline plutus-core/plutus-ir/src/PlutusIR/Compiler.hs:146]: This is the only Haskell file changed and this line is the idea of this PR.

**Comment by effectfully** [comment]
I'll merge the PR just to unblock people, but please do still review it.

File: plutus-core/plutus-ir/src/PlutusIR/Compiler.hs (hunk 1 above)
Comment (ana-pantilie) [reply plutus-core/plutus-ir/src/PlutusIR/Compiler.hs:146]: So only the PIR case-of-case has this problem, the UPLC one doesn't?

**Review by ana-pantilie — COMMENTED** [review]
Sorry, I need to clean up my GitHub notifications so I just saw this one.

File: plutus-core/plutus-ir/src/PlutusIR/Compiler.hs (hunk 1 above)
Comment (effectfully) [reply plutus-core/plutus-ir/src/PlutusIR/Compiler.hs:146]: That's a great question, thank you for asking it! Yes, I believe UPLC case-of-case has one of the sources of exponentiality of PIR case-of-case. I've amended #5966 to reflect that.

Comment (Unisay) [reply plutus-core/plutus-ir/src/PlutusIR/Compiler.hs:146]: What about the corresponding CLI flag - Should we remove it too?

**Pull request merged**